activerecord/lib/active_record/persistence.rb

```diff
@@ -460,19 +463,18 @@ def reload(options = nil)
     #   ball = Ball.new
     #   ball.touch(:updated_at)   # => raises ActiveRecordError
     #
-    def touch(*names)
+    def touch(*names, time: current_time_from_proper_timezone)
       raise ActiveRecordError, "cannot touch on a new record object" unless persisted?
 
       attributes = timestamp_attributes_for_update_in_model
       attributes.concat(names)
 
       unless attributes.empty?
-        current_time = current_time_from_proper_timezone
```

Comment:
It's easier to use the time argument and reassigning it to current_time doesn't improve clarity :wink: